### DISCUSSION TITLE
How to build custom driver (like fake-driver example)?

### DISCUSSION THREAD

**Discussion by youngfreeFJS** [appium/appium#20643] [Q&A]
I downloaded the source code from the repo and navigated to the fake-driver folder. I wanted to install the fake driver globally for Appium (CLI), so I followed these steps:

Run npm i in the ~/codes/appium/packages/fake-driver directory.
Then, run npm run build.
After that, I executed this command ```appium driver install --source=local ~/codes/appium/packages/fake-driver```

However, the installation failed with the following error:
```
appium git:(master) appium driver install --source=local  '/Users/youngfreefjs/Desktop/code/appium/packages/fake-
driver'
✔ Installing '/Users/youngfreefjs/Desktop/code/appium/packages/fake-driver'
Error: ✖ driver "fake" had error and will not be available:
  - Unable to register schema at path ./build/lib/fake-driver-schema.js; Cannot find module '@appium/fake-driver/build/lib/fake-driver-schema.js'
Require stack:
- /Users/youngfreefjs/.appium/noop.js (Actual value: "./build/lib/fake-driver-schema.js")
```

I noticed that there was only one image in the build path, and the command in pkg.json is: ```"build": "cpy lib/screen.png build"```.
This seems like a fake build command. How can I do a real build in the development process?

**Comment by KazuCocoa** [comment]
for the fake driver's case, maybe you need to run npm install once in the appium repository's root as lerna managed repo.
Or if you want only https://www.npmjs.com/package/@appium/fake-driver package:

1. install it via npm with `apium driver install --source=npm @appium/fake-driver`, or
2. `npm pack @appium/fake-driver` and unpack the downloaded package, then `appium driver install --source=local /path/to/the/dir`
